### ISSUE TITLE
[BUG]: no way to append to a list, returning a new list with the added value

### ISSUE THREAD

**Issue by paultag** [KittyCAD/modeling-app#4127]
### Describe the bug

I would like to be able to append to a list -- something like:

```
fn x = (something) => {
    return something.append(1)
}

const y = x([3, 2])
```

### Steps to Reproduce

n/a

### Expected Behavior

n/a

### Desktop OS

Debian

### Browser

Chrome

### Version

main

**Comment by nrc** [comment]
What's your use case for this? I'm curious about the larger task you're doing that requires it

**Comment by jtran** [comment]
#4232 implements this by the most straight forward approach of cloning the Vec and pushing to it.

This technically works, but if anyone pushes inside a loop, it's O(n^2). Do we really want to introduce this? Seems like a big foot-gun.

As an alternative, I propose persistent data structures since they fit better with immutability. Clojure makes good use of hash array mapped tries to provide a persistent interface, while mutating internally for better performance.

**Comment by jtran** [comment]
Resolved by #4341.
